client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/classes/FloatClassReplacement.java

```diff
@@ -0,0 +1,96 @@
+package org.evomaster.client.java.instrumentation.coverage.methodreplacement.classes;
+
+
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.MethodReplacementClass;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.Replacement;
+import org.evomaster.client.java.instrumentation.heuristic.Truthness;
+import org.evomaster.client.java.instrumentation.shared.ReplacementType;
+import org.evomaster.client.java.instrumentation.shared.StringSpecialization;
+import org.evomaster.client.java.instrumentation.shared.StringSpecializationInfo;
+import org.evomaster.client.java.instrumentation.shared.TaintInputName;
+import org.evomaster.client.java.instrumentation.staticstate.ExecutionTracer;
+
+import static org.evomaster.client.java.instrumentation.coverage.methodreplacement.DistanceHelper.*;
+
+public class FloatClassReplacement implements MethodReplacementClass {
+
+    @Override
+    public Class<?> getTargetClass() {
+        return Float.class;
+    }
+
+
+    @Replacement(type = ReplacementType.EXCEPTION, replacingStatic = true)
+    public static float parseFloat(String input, String idTemplate) {
+
+        if (TaintInputName.isTaintInput(input)) {
+            ExecutionTracer.addStringSpecialization(input,
+                    new StringSpecializationInfo(StringSpecialization.INTEGER, null));
+        }
+
+        if (idTemplate == null) {
+            return Float.parseFloat(input);
+        }
+
+        try {
+            float res = Float.parseFloat(input);
+            ExecutionTracer.executedReplacedMethod(idTemplate, ReplacementType.EXCEPTION, new Truthness(1, 0));
+            return res;
+        } catch (NumberFormatException | NullPointerException e) {
+            double h = parseFloatHeuristic(input);
+            ExecutionTracer.executedReplacedMethod(idTemplate, ReplacementType.EXCEPTION, new Truthness(h, 1));
+            throw e;
+        }
+    }
+
+    /**
+     * Optimizes for Java regex pattern "['-']?[0-9]*['.']?[0-9]*"
+     * @param input
+     * @return
+     */
+    protected static double parseFloatHeuristic(String input) {
+
+        if (input == null) {
+            return H_REACHED_BUT_NULL;
+        }
+
+        final double base = H_NOT_NULL;
+
+        if (input.length() == 0) {
+            return base;
+        }
+
+        long distance = 0;
+
+        if (input.length() == 1) {
+            //cannot be '-'
+            distance += distanceToDigit(input.charAt(0));
+        } else {
+
+            for (int i = 0; i < input.length(); i++) {
```

Comment:
isn't there a certain number of digits that a float/double can have? what happen in Java if a very long string is given as input? would `parseDouble()` round the number? or throw an exception? could add some test cases to validate it, eg with "0.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001"

Reply:
If the value is too small, it is truncated. If the value is too big, it goes to POSITIVE_INFINITY